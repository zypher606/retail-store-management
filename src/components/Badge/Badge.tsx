import React from 'react';
import { createStyles, Theme, makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles((theme: Theme) => 
  createStyles({
    folderBadge: {
      color: '#fff',
      fontSize: '12px',
      padding: '1px 7px',
      borderRadius: '2px',
      display: 'inline',
    },
  })
);

interface IBadge {
  color: string;
  children: any;
}
export const Badge = ({
  color: background,
  children,
  ...rest
}: IBadge)  => {

  const classes = useStyles();

  return (
    <div data-testid="badge-container" {...rest} className={classes.folderBadge} style={{background}}>{children}</div>
  )
}