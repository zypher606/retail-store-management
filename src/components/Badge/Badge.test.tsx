import React from 'react';
import ReactDOM, { unmountComponentAtNode } from 'react-dom';
import { isTSAnyKeyword } from '@babel/types';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";
import '@testing-library/jest-dom';
import { v4 as uuidv4 } from 'uuid';
import { Badge } from './Badge';

afterEach(() => {
  // cleanup on exiting
  cleanup();
});

describe("Badge component tests", () => {
  it("rendered successful", () => {
    const div = document.createElement("div");
    render(
      <Badge color="red">Hello world with my custom children</Badge>
    );
  });
  
  it('renders with props correctly', () => {
    const children = 'Hello world in here';
    const { getByTestId } = render(
      <Badge color="red">{children}</Badge>
    );
    expect(getByTestId("badge-container")).toHaveTextContent(children);
  });


  it('matches snapshot', () => {
    const tree = renderer.create(
      <Badge color="#efe112">
        <div>
          Custom badge in here
        </div>
        <p>Another child</p>
      </Badge>
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
  
});