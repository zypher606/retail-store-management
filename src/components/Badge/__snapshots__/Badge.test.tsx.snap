// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component tests matches snapshot 1`] = `
<div
  className="makeStyles-folderBadge-3"
  data-testid="badge-container"
  style={
    Object {
      "background": "#efe112",
    }
  }
>
  <div>
    Custom badge in here
  </div>
  <p>
    Another child
  </p>
</div>
`;
